src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/PKCS1MaskGenerationMethod.cs

```diff
@@ -18,19 +18,13 @@ public PKCS1MaskGenerationMethod()
         public string HashName
         {
             get { return _hashNameValue; }
-            set 
-            {
-                if (value != null && value != "SHA1")
-                {
-                    throw new PlatformNotSupportedException();
-                }
-            }
+            set { _hashNameValue = value ?? "SHA1"; }
```

Comment:
Oh, I see.  I looked straight at it and thought it was assigning SHA1.  ::whistles innocently::